src/interfaces/event-observable.interface.ts

```diff
@@ -1,5 +1,6 @@
 import { Observable } from 'rxjs';
+import { Type } from '@nestjs/common';
 
 export interface EventObservable<T> {
-  ofType(...events): Observable<T>;
+  ofType(...events: Type<T>[]): Observable<T>;
```

Comment:
generic type `<T>` should be different for both (1) dispatched (2) returned observable. I'd suggest sth like:
```
EventObservable<TInput, TOutput>
```